modules/simple_forms_api/app/services/simple_forms_api/notification_email.rb

```diff
@@ -140,6 +138,19 @@ def send_email_now(template_id, data)
       end
     end
 
+    def get_first_name
+      if user_account
+        mpi_profile = MPI::Service.new.find_profile_by_identifier(identifier_type: 'ICN', identifier: user_account.icn)
+        if mpi_profile
+          raise mpi_profile.error if mpi_profile.error
+
+          mpi_profile.first_name
```

Comment:
What about adding a safeguard here if mpi_profile.first_name is nil? You could do something like:
```
      return mpi_profile.first_name if mpi_profile.first_name
      
      # Then raise an exception or log if first_name is missing
      raise "First name not found in MPI profile"
```